### ISSUE TITLE
Nearest without value

### ISSUE THREAD

**Issue by cpacia** [OpenBazaar/OpenBazaar-Server#86]
Section 2.3 of the kademlia requires that node's who look up a value using a FIND_VALUE query issue a STORE request to the nearest node that did not have the value. 

The purpose of this is to cache the data further away from that neighborhood, reducing the time it takes for future look ups. But to prevent the caching from getting out of control (replicated on almost every node), nodes outside the neighborhood are supposed to set the ttl exponentially inversely proportional to the number of nodes between the current node and the node whose ID is closet to the key.

The problem is that our code base currently implements caching the data at the nearest without value, but it doesn't implement the reduced ttl (meaning the data may end up replicated on all nodes). So basically when we receive a STORE request we need to check our distance to that neighborhood and adjust the ttl properly before storing it. 

This could also create a problem with our republish function. If we republish keys with this reduced ttl, and that STORE request makes it to a new node without the value, that node will store the reduced ttl instead of the full ttl it should be storing. Maybe we need to set the ttl back to it's full value before republishing (if that's possible) or not republish keys outside our neighborhood.

**Comment by gubatron** [comment]
@cpacia but you're asking only THE nearest node. Why would it get out of hand? will that node will also perform a find?

**Comment by cpacia** [comment]
Well if 1000 people search for "shoes" then each query another node is told to store the value. That could be 1000 nodes outside the neighborhood where "shoes" resides.

**Comment by gubatron** [comment]
yeah, but that's not going to be the only search going through, and eventually your buckets should be filled to the point that those nodes that are not in your neighborhood can't be stored, right?
